compiler/rustc_hir_analysis/src/check/fn_ctxt/suggestions.rs

```diff
@@ -80,6 +80,38 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let Some((def_id_or_name, output, inputs)) = self.extract_callable_info(expr, found)
             else { return false; };
         if can_satisfy(output) {
+            // suggest `unwrap_or_else` for `Option` and `Result`, when a closure is given
+            if let hir::ExprKind::Closure(..) = expr.kind &&
+                let parent = self.tcx.hir().get_parent_node(expr.hir_id) &&
+                let Some(Node::Expr(call_expr)) = self.tcx.hir().find(parent) &&
+                let hir::ExprKind::MethodCall(
+                      hir::PathSegment {ident: method_name, ..},
+                      receiver,
+                      _args,
+                      _span) = call_expr.kind &&
+                inputs.len() == 0 {
+              let self_ty = self.typeck_results.borrow().expr_ty(receiver);
+              let new_name = Ident {
+                  name: Symbol::intern(&format!("{}_else", method_name.as_str())),
+                  span: method_name.span,
+              };
+
+              if self.method_exists(
+                  new_name,
+                  self_ty,
+                  call_expr.hir_id,
+                  false,
+              ) {
+                  err.span_suggestion_verbose(
+                      method_name.span.shrink_to_hi(),
+                      &format!("try calling `{}` instead", new_name.name.as_str()),
+                      "_else",
+                      Applicability::MachineApplicable,
```

Comment:
This isn't machine applicable...? It's very possible for the suggestion to be wrong :sweat_smile:

Reply:
Fixed!